internal/service/static_ip/resource_static_ip.go

```diff
@@ -98,19 +97,35 @@ func resourceStaticIPCreate(ctx context.Context, d *schema.ResourceData, m inter
 	return resourceStaticIPRead(ctx, d, m)
 }
 
-func resourceStaticIPDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceStaticIPDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(*aiven.Client)
 
 	project, staticIPAddressId := schemautil.SplitResourceID2(d.Id())
 
-	err := client.StaticIPs.Delete(
+	staticIP, err := client.StaticIPs.Get(project, staticIPAddressId)
+	if err != nil && !aiven.IsNotFound(err) {
+		return diag.FromErr(err)
+	} else {
+		if staticIP.State == schemautil.StaticIpAssigned || staticIP.State == schemautil.StaticIpAvailable {
+			if err := client.StaticIPs.Dissociate(project, staticIPAddressId); err != nil {
+				return diag.FromErr(err)
+			}
+
+			if err := resourceStaticIPWait(ctx, d, m); err != nil {
+				return diag.FromErr(fmt.Errorf("unable to wait for static ip to become unassigned: %w", err))
```

Comment:
Pls use `diag.Errorf` instead of `diag.FromErr(fmt.Errorf(`

Reply:
I did it this way for consistence purposes because it's made this way in `resourceStaticIPCreate` too